students/jarosz_natalia/lesson_10/filling_in_the_gaps.py

```diff
@@ -0,0 +1,36 @@
+import re
+import os
+import shutil
+
+pattern = 'spam'
+path = '../lesson_10/test/'
+
+
+def main():
+    sorted_numbers = []
+    prefix_pattern = re.compile(r"{0}\d{{3}}".format(pattern))
+
+    for foldername, subfolder, filenames in os.walk(path):
+        for filename in filenames:
+            result = re.search(prefix_pattern, filename)
+            if result is not None:
+                sorted_numbers.append(result.group(0).replace(pattern, ''))
+
+    sorted_numbers = sorted(sorted_numbers)
+
+    for i in range(int(sorted_numbers[0]), int(sorted_numbers[-1]) - 1):
```

Comment:
It is working but for easiest cases when we have two files spam001.txt and spam003.txt. What is the case when in test folder we will have spam001.txt, spam003.txt and spam005.txt?